internal/pkg/csumio/csumreader_test.go

```diff
@@ -184,3 +184,226 @@ func TestReader_CorrectChecksum_VerificationEnabled(t *testing.T) {
 		})
 	}
 }
+
+// TestReader_ReadFactorCoverage exhaustively validates csumio.Reader.Read.
+// The test matrix uses pairwise coverage across three factors:
+//
+//	Factor A: requested len(p) relative to chunkSize (> chunk / == chunk / < chunk).
+//	Factor B: remaining unread bytes in `dataReader` relative to len(p) (> p / == p / < p).
+//	Factor C: inner-buffer state (empty vs. partially filled by a prior read).
+//
+// These three factors yield nine cases (case01–case09). Each entry below sets the total payload size,
```

Comment:
Now all cases are just 18. How about implement all cases than part of cases by using pairwise coverage? Yes, I said "let's use pairwise coveray" at the first time. So I can add the remaining cases later.